test-report/src/main/java/io/opentelemetry/instrumentation/testreport/FlakyTestReporter.java

```diff
@@ -80,6 +88,28 @@ private void scanTestFile(Path testReport) {
       return;
     }
 
+    // google sheets don't automatically recognize dates with time zone, here we reformat the date
+    // so that it wouldn't have the time zone
+    TemporalAccessor ta =
+        DateTimeFormatter.ISO_DATE_TIME.parseBest(
+            timestamp, ZonedDateTime::from, LocalDateTime::from);
+    timestamp = DateTimeFormatter.ISO_LOCAL_DATE_TIME.format(ta);
+
+    // when test result file was modified more than 20 minutes after the time in the results file
+    // then the test results were restored from gradle cache and the test wasn't actually executed
+    Instant reportModified = Instant.ofEpochMilli(testReport.toFile().lastModified());
+    reportModified = reportModified.minus(20, ChronoUnit.MINUTES);
+    Instant testExecuted = null;
+    if (ta instanceof ZonedDateTime) {
+      testExecuted = ((ZonedDateTime) ta).toInstant();
+    } else if (ta instanceof LocalDateTime) {
+      testExecuted = ((LocalDateTime) ta).toInstant(OffsetDateTime.now().getOffset());
+    }
+    if (testExecuted != null && reportModified.isAfter(testExecuted)) {
+      System.err.println(
+          "Ignoring " + testReport + " since it appears to be restored from gradle build cache");
```

Comment:
should we return here, or is it enough to log when it occurs?

Reply:
thanks, added the missing return